stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -363,9 +363,53 @@ bool RecordTypeInfo::readExtraInhabitantIndex(remote::MemoryReader &reader,
     return false;
   }
 
-  case RecordKind::MultiPayloadEnum:// XXX TODO
-    return false;
-    
+  case RecordKind::MultiPayloadEnum: {
```

Comment:
This block of code counts XIs in MultiPayloadEnums and returns the digested result for an enclosing single-payload enum to use.